### ISSUE TITLE
Blog für Crowdgezwitscher

### ISSUE THREAD

**Issue by Bensk1** [Strassengezwitscher/Strassengezwitscher#178]
Crowdgezwitscher soll einen typischen Blog bekommen. Im Betreiberbereich kann ein Eintrag hinzugefügt werden, der im Nutzerbereich angezeigt wird.

**Comment by mathebox** [comment]
Ich bin mir gerade unschlüssig, ob wir das mit Django oder Angular implementieren sollten.

**Django**
- Pro:
  - Wohl die einfachste Variante mit den Django Templates zu bauen
- Con:
  - Das Design an die Angular-App anzupassen ist schwierig.
  - Keine Ahnung wie gut das mit dem Angular Routing (und Ladezeiten funktioniert)

**Angular**
- Pro:
  - Einheitliches Design für den User-Bereich
- Con:
  - Blogeinträge müssen über Rest API ausgeliefert werden.

Es darf gerne ergänzt werden.
Was denkst du @piepmatz?

**Comment by piepmatz** [comment]
Ich würde zuallererst über "implementieren" diskutieren. So ein Datenmodell für Beiträge kriegen wir sicherlich schnell hin, aber dank Wordpress und Co. sind die Erwartungen an ein Blog vermutlich so groß, dass wir allein in den Blog viel Zeit stecken könnten.
Wir sollten also Rücksprache mit @humuzz halten, was die Anforderungen ans Blog sind.
Sollte sich das als umfangreich herausstellen, würde ich dafür votieren, dass wir da möglichst wenig selbst machen und uns nach einer fertigen Lösung umschauen, die wir leicht integriert und angepasst (Design) bekommen. Vorzugsweise als Django-App, aber andere Formate, vielleicht auch für Angular, kann man sicher diskutieren.

**Comment by Bensk1** [comment]
Ich hatte mit Alexej gesprochen und die Anforderungen in die Tickets einfließen lassen. Es soll ein einfacher Blog werden mit Überschrift, formatiertem Text, Bildern im Text, einem Autor, Datum und **ohne** Kommentare. Das muss nicht unbedingt implementiert werden, aber ich würde gern einen dritten Account, neben Betreiberbereich und Mattermost vermeiden wollen.

**Comment by piepmatz** [comment]
Dritten Account würde ich auch vermeiden wollen. Zinnia (in verlinkter Übersicht Spalte 3) scheint Djangos mitgeliefertes Authentication-System zu verwenden. Das machen wir ja auch. Das sollte einen gemeinsamen Login nicht allzu schwierig machen.

Wenn man es selbst implementieren wollte: Die Anforderungen klingen ja noch ganz gut machbar. TinyMCE kann das alles. Dann bräuchten wir lediglich ein Model für einen Blogbeitrag mit einem Feld für das HTML des Beitrags, evtl. ein Titel-Feld, eine Referenz auf User, ein Datumsfeld und irgendwo müssten wir die hochgeladenen Bilder wegspeichern.

**Comment by Bensk1** [comment]
Genau, so kompliziert klingt das nicht. Ich hätte auch nix gegen Zinnia, was auch immer ihr für einfacher haltet.

**Comment by humuzz** [comment]
Eine Frage: FB-Seiten bzw. Veranstaltungen sollen ja auch eigene Seiten mit dem Twitter-Feed und Sonstigen Infos bekommen. Sind das dann vom Aufbau her die gleichen Seiten/Artikel wie im Blog oder werden die komplett unterschiedlich erstellt?

**Comment by piepmatz** [comment]
Mein persönliches Verständnis ist wie folgt: Die zusätzlichen Infos zu einer FB-Seite bzw. Veranstaltung werden in einem zusätzlichen Bereich über die Karte gelegt. Was dort angezeigt wird, sind vor allem Details, also Links, Veranstalter, Motto, Teilnehmer usw. Dazu noch der Twitter-Feed.
Blogposts hingegen stelle ich mir in der Erstellung als ein großes Freitextfeld vor, in dem eben geschrieben, Text formatiert, Bilder eingefügt werden können usw. Dort habt ihr also beim Verfassen Einfluss auf die Gestaltung. Weil die Blogposts umfangreicher werden könnten als die zusätzlichen Informationen zu Objekten auf der Karte, würde ich sie nicht bei der Karte anzeigen, sondern ihnen einen eigenen Bereich, beispielsweise eine Art Tab, einräumen.

**Comment by humuzz** [comment]
Okay. Ich überlege nur, wie man diese Bereiche zu den einzelnen FB-Seiten und Veranstaltungen, die über der Karte liegen, mit relevanten Blogeinträgen verknüpft. So wie z.B. die FB-Seite und Veranstaltungen miteinander über "Verbundene Veranstaltungen" bzw. "Veranstaltende Organisationen" verknüpft sein sollten. Wahrscheinlich ist die automatische Zuordnung von Blogposts aber zu aufwändig (z.B. über Schlagwörter). Wenn aber man z.B. in dem Info-Bereich noch manuell  "Verwandte Blogeinträge" mit den zugehörigen Links darunter ergänzen könnte, wäre das schon nicht schlecht.

**Comment by piepmatz** [comment]
Das sollte sich mal lassen.

**Comment by mathebox** [comment]
Wenn das so miteinander verknüpft sein soll, dann würde es sich ja anbieten das selber zu implementieren, oder?

**Comment by piepmatz** [comment]
Naja, dem Datenmodell einer bestehenden Lösung noch ein Feld hinzuzufügen ist auch kein Problem.
Ich habe momentan keine Präferenz, wie wir das umsetzen.

**Comment by humuzz** [comment]
Ich weiß nicht wie aufwändig das Implementieren ist. Ehe ihr da jetzt einen neuen riesen Brocken abarbeiten müsst deswegen, nehmen wir die manuelle Lösung.

**Comment by mathebox** [comment]
Da der Blog unter dem Name Crowdgezwitscher laufen soll würde ich folgendes vorschlagen:
- Verwendung einer 3rd-Party-Django-App
  - um uns den Implementieraufwand im Backend zu ersparen
  - wird ja sicherlich auch mit einem Editor kommen
- Anzeige via Angular
  - damit das Design nicht bricht und keine "Ladebildschirm" beim wechseln zwischen Blog und Karte zu sehen ist (in unserem Fall das Crowdgezwitscher-Logo)
  - daher auch die Auslieferung von Blogeinträge via REST framework

Okay so? Habt ihr Anmerkungen? @piepmatz
